src/utils/bibModelPreprocessing.ts

```diff
@@ -0,0 +1,101 @@
+import type { SearchResult, Note } from "../types/searchTypes"
+
+const getGroupedNotes = (bib: SearchResult) => {
+  const note = bib?.note?.length ? bib.note : null
+  let notesGroupedByNoteType = {}
+
+  /**
+   * getNoteType(note)
+   * Construct label for a note by adding the word 'Note'
+   */
+  const getNoteType = (note: Note) => {
+    const type = note.noteType || ""
+    return type.toLowerCase().includes("note") ? type : `${type} (note)`
+  }
+
+  if (!note) {
+    return notesGroupedByNoteType
```

Comment:
You can move this towards the top of the function and return an empty object when it's not set. This makes it more legible/explicit since I think this implies that notesGroupedByNoteType contains something other than an empty object at this point.

Reply:
It also might not need to be instantiated at all if you return the note.filter.reduce statement.